src/sentry/api/issue_search.py

```diff
@@ -73,7 +74,15 @@ def visit_boolean_operator(self, node, children):
 
 
 def parse_search_query(query):
-    tree = event_search_grammar.parse(query)
+    try:
+        tree = event_search_grammar.parse(query)
+    except IncompleteParseError as e:
+        raise InvalidSearchQuery(
+            '%s %s' % (
+                u'Parse error: %r (column %d).' % (e.expr.name, e.column()),
+                'This is commonly caused by unmatched-parentheses.',
```

Comment:
Wondering if it'd be helpful to suggest that they enclose the text part of the statement in quotes